clients/cli/src/proto/nexus.orchestrator.rs

```diff
@@ -67,6 +67,16 @@ pub struct SubmitProofRequest {
     /// The task's ID.
     #[prost(string, tag = "6")]
     pub task_id: ::prost::alloc::string::String,
+    /// A Ed25519 public key (~32 bytes) generated by
+    /// the node to uniquely identify this request,
+    /// used for DoS protection.
+    #[prost(bytes = "vec", tag = "7")]
+    pub ed25519_public_key: ::prost::alloc::vec::Vec<u8>,
```

Comment:
Can you explain this more? How is this used to verify the signature? And what is it intended to protect against?

Generally speaking in-band signaling like this will almost always provide a way for an adversary to undercut the security guarantees this is intended to provide.